### PR TITLE
Reformatted files to match more generalised code patterns, and adding a few miscellaneous/game commands.

### PR DESCRIPTION
**Pull request by Shlol762**

### CODE REVIEW

File: cogs/etc.py [hunk 1]
```diff
@@ -395,5 +402,12 @@ def to_string(c):
             return await ctx.send('Output too long to display.')
         await ctx.send(msg)
 
-def setup(bot):
-    bot.add_cog(Etc())
+    @commands.command(name="Profilepic", aliases=['pfp'])
```
Comment (TAG-Epic) [inline cogs/etc.py:405]: I really dont see the reason for this to be added

File: cogs/etc.py [hunk 2]
```diff
@@ -374,13 +374,20 @@
...
 import unicodedata
-from discord.ext import commands
+
+import nextcord
+from nextcord.ext import commands
 
 class Etc(commands.Cog):
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.name = "Etectera(etc)"
```
Comment (TAG-Epic) [inline cogs/etc.py:386]: everyone understands what etc means

File: cogs/help.py [hunk 3]
```diff
@@ -38,27 +52,85 @@ async def create_view(self):
...
-    @commands.command()
+    @commands.command(hidden=True)
     @commands.is_owner()
-    async def help_menu(self, ctx):
+    async def help_menu(self, ctx: commands.Context):
         await ctx.send("Click a button to create a help thread!", view=HelpView())
 
-    @commands.command()
-    async def close(self, ctx):
+    @commands.command(name="Close", aliases=['cs'])
```
Comment (TAG-Epic) [inline cogs/help.py:64]: Why is commands uppercase now

File: cogs/help.py [hunk 4]
```diff
@@ -38,27 +52,85 @@ async def create_view(self):
...
-        if help_role in ctx.author.roles or history_flat[0].mentions[0].id == ctx.author.id:
-            if isinstance(ctx.channel, nextcord.Thread) and ctx.channel.parent_id == self.help_channel:
-                await ctx.send("This thread has now been closed. Please create another thread if you wish to ask another question.")
+        if isinstance(ctx.channel, Thread) and ctx.channel.parent_id == self.help_channel:
+            history_flat = await history.flatten()
+            log_channel: TextChannel = get(ctx.guild.channels, name='help_logs')
+            help_role: Role = find(lambda role: role.name == 'My help is bad', ctx.message.guild.roles)
+            if help_role in ctx.author.roles or history_flat[0].mentions[0].id == ctx.author.id:
+                await ctx.send(
+                    "This thread has now been closed. Please create another thread if you wish to ask another question.")
```
Comment (TAG-Epic) [inline cogs/help.py:73]: This looks bad

File: cogs/help.py [hunk 5]
```diff
@@ -38,27 +52,85 @@ async def create_view(self):
...
                 await ctx.channel.edit(locked=True, archived=True)
-                await log_channel.send(f"Help thread {ctx.channel.name} (created by {history_flat[0].mentions[0].name}) has been closed.")
+                await log_channel.send(
+                    f"Help thread {ctx.channel.name} (created by {history_flat[0].mentions[0].name}) has been closed.")
+
+    @commands.command(name='Help', aliases=['hp'],
+             help="Displays the commands of me.",
+             usage="help|hp (command|command alias|category|category alias)")
+    async def help(self, ctx: Context, *, arg: Optional[str]):
```
Comment (TAG-Epic) [inline cogs/help.py:81]: Optional typehints doesnt work with no defaults.

File: cogs/help.py [hunk 6]
```diff
@@ -38,27 +52,85 @@ async def create_view(self):
...
+            help_role: Role = find(lambda role: role.name == 'My help is bad', ctx.message.guild.roles)
+            if help_role in ctx.author.roles or history_flat[0].mentions[0].id == ctx.author.id:
+                await ctx.send(
+                    "This thread has now been closed. Please create another thread if you wish to ask another question.")
                 await ctx.channel.edit(locked=True, archived=True)
-                await log_channel.send(f"Help thread {ctx.channel.name} (created by {history_flat[0].mentions[0].name}) has been closed.")
+                await log_channel.send(
+                    f"Help thread {ctx.channel.name} (created by {history_flat[0].mentions[0].name}) has been closed.")
+
+    @commands.command(name='Help', aliases=['hp'],
```
Comment (TAG-Epic) [inline cogs/help.py:78]: imo using MinimalHelpCommand would be better here.

Comment (TAG-Epic) [reply cogs/help.py:78]: This helpcommand doesnt subclass the default help command.

File: cogs/rtfm.py [hunk 7]
```diff
@@ -381,10 +381,10 @@
 import zlib
 from typing import Dict
 
-import nextcord as discord
+import nextcord
 from nextcord.ext import commands
 
-from .utils import fuzzy
+from utils import fuzzy
```
Comment (TAG-Epic) [inline cogs/rtfm.py:387]: This doesnt work.

File: cogs/rtfm.py [hunk 8]
```diff
@@ -532,12 +534,12 @@ async def rtfm_group(self, ctx: commands.Context, *, obj: str = None):
     async def rtfm_python_cmd(self, ctx: commands.Context, *, obj: str = None):
         await self.do_rtfm(ctx, "python", obj)
 
-    @commands.command(help="delete cache of rtfm (owner only)", aliases=["purge-rtfm", "delrtfm"])
+    @commands.command(hidden=True, help="delete cache of rtfm (owner only)", aliases=["purge-rtfm", "delrtfm"])
```
Comment (TAG-Epic) [inline cogs/rtfm.py:537]: This shouldnt be hidden???

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: cogs/rtfm.py (hunk 8 above)
Comment (Shlol762) [reply cogs/rtfm.py:537]: oh aight

File: cogs/help.py (hunk 6 above)
Comment (Shlol762) [reply cogs/help.py:78]: well no idk how to...

File: cogs/rtfm.py (hunk 7 above)
Comment (Shlol762) [reply cogs/rtfm.py:387]: hm ide ran it without problems

File: cogs/help.py (hunk 5 above)
Comment (Shlol762) [reply cogs/help.py:81]: Oh, sorry ill keep that in mind,

File: cogs/help.py (hunk 3 above)
Comment (Shlol762) [reply cogs/help.py:64]: thats how the help command works. i mean if you want it to be lowercase then lower case ig

File: cogs/etc.py (hunk 2 above)
Comment (Shlol762) [reply cogs/etc.py:386]: well yea true

File: cogs/etc.py (hunk 1 above)
Comment (Shlol762) [reply cogs/etc.py:405]: why dont they have a /shrug on github lmao. simply

File: cogs/help.py (hunk 3 above)
Comment (TAG-Epic) [reply cogs/help.py:64]: That would require you to use =Close...?

Comment (Shlol762) [reply cogs/help.py:64]: wait you havent set it to `case_insenstive=True`??

Comment (TAG-Epic) [reply cogs/help.py:64]: no

File: cogs/etc.py (hunk 2 above)
Comment (ooliver1) [reply cogs/etc.py:386]: change it to just etc then

**Comment by ooliver1** [comment]
looks like you have resolved things that arent resolved

File: cogs/etc.py [hunk 9]
```diff
@@ -374,13 +374,20 @@
...
+        self.description = "Random stuff... 'Etcetera'!"
```
Comment (ooliver1) [inline cogs/etc.py:387]: also maybe just etc

File: cogs/etc.py [hunk 10]
```diff
@@ -395,5 +402,12 @@ def to_string(c):
             return await ctx.send('Output too long to display.')
         await ctx.send(msg)
 
-def setup(bot):
-    bot.add_cog(Etc())
+    @commands.command(name="Profilepic", aliases=['pfp'])
+    async def _pfp(self, ctx: commands.Context, user: Optional[User]):
+        user = await commands.UserConverter().convert(ctx, user) if user else ctx.author
+        user = await self.bot.fetch_user(user.id)
+        await ctx.send(embed=nextcord.Embed(description='_ _', type="ok", colour=""))
```
Comment (ooliver1) [inline cogs/etc.py:409]: this looks like it never uses this ``user`` var

File: cogs/help.py [hunk 11]
```diff
@@ -1,35 +1,49 @@
...
+    @button(label="D.py libs help", style=ButtonStyle.gray, custom_id="help:dpylibs")
+    async def nextcord_lib_help(self, _button: Button, interaction: Interaction):
         await self.create_help_thread("Discord.py libraries", interaction)
-    
+
     # Python
-    @ui.button(label="Python help", style=ButtonStyle.green, custom_id="help:python")
-    async def python_help(self, _, interaction: Interaction):
+    @button(label="Python help", style=ButtonStyle.green, custom_id="help:python")
+    async def python_help(self, _button: Button, interaction: Interaction):
```
Comment (ooliver1) [inline cogs/help.py:27]: its just `_` as its never used

File: cogs/help.py [hunk 12]
```diff
@@ -1,35 +1,49 @@
...
-    async def create_help_thread(self, name, interaction):
-        thread = await interaction.channel.create_thread(name=f"{name} help ({interaction.user})", type=ChannelType.public_thread)
+
+    async def create_help_thread(self, name: str, interaction: Interaction):
+        thread: Thread = await interaction.channel.create_thread(name=f"{name} help ({interaction.user})",
+                                                                 type=ChannelType.public_thread)
         await interaction.response.send_message("Created!", ephemeral=True)
         await thread.add_user(interaction.user)
-        log_channel = nextcord.utils.get(interaction.guild.channels, name='help_logs')
+        log_channel: TextChannel = get(interaction.guild.channels, name='help_logs')
```
Comment (ooliver1) [inline cogs/help.py:35]: get on guild.text_channels maybe?

File: cogs/help.py (hunk 3 above)
Comment (ooliver1) [reply cogs/help.py:64]: this was never resolved but has been marked as so

File: cogs/help.py [hunk 13]
```diff
@@ -38,27 +52,85 @@ async def create_view(self):
...
-        history_flat = await history.flatten()
-        log_channel = nextcord.utils.get(ctx.guild.channels, name='help_logs')
-        help_role = nextcord.utils.find(lambda r: r.name == 'My help is bad', ctx.message.guild.roles)
-        if help_role in ctx.author.roles or history_flat[0].mentions[0].id == ctx.author.id:
-            if isinstance(ctx.channel, nextcord.Thread) and ctx.channel.parent_id == self.help_channel:
-                await ctx.send("This thread has now been closed. Please create another thread if you wish to ask another question.")
+        if isinstance(ctx.channel, Thread) and ctx.channel.parent_id == self.help_channel:
+            history_flat = await history.flatten()
+            log_channel: TextChannel = get(ctx.guild.channels, name='help_logs')
+            help_role: Role = find(lambda role: role.name == 'My help is bad', ctx.message.guild.roles)
```
Comment (ooliver1) [inline cogs/help.py:70]: get_role would probbaly be better, also utils.get does the same

File: cogs/help.py (hunk 6 above)
Comment (ooliver1) [reply cogs/help.py:78]: then why change it

File: cogs/help.py (hunk 5 above)
Comment (ooliver1) [reply cogs/help.py:81]: you seem to have marked a lot as resolved and not done anything

File: main.py [hunk 14]
```diff
@@ -3,36 +3,38 @@
...
-from nextcord import Intents
+from nextcord import Intents, Message
 from nextcord.ext import commands
 from nextcord.ext.commands import errors
 
 intents = Intents.all()
 intents.presences = False
 
 bot = commands.Bot("=", intents=intents)
+bot.remove_command("help")
```
Comment (ooliver1) [inline main.py:14]: :notlikeduck:

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Comment by TAG-Epic** [comment]
No updates in 2 months. Feel free to reopen if you want to continue this

**Pull request closed without merging**